balboa-client-kafka/src/main/scala/com/socrata/metrics/producer/GenericKafkaProducer.scala

```diff
@@ -0,0 +1,71 @@
+package com.socrata.metrics.producer
+
+import java.util.Properties
+import com.socrata.balboa.common.kafka.Constants
+import com.socrata.balboa.common.kafka.util.AddressAndPort
+import kafka.producer.{KeyedMessage, Producer, ProducerConfig}
+import kafka.serializer.Encoder
+
+/**
+ * A wrapper around Kafka [[Producer]].  This Producer is restricted to exactly one Kafka Topic.  This
+ * producers responsibility is to provide many of the default properties for Kafka [[Producer]].  To see the
+ * current default settings reference [[Constants.producerDefaultProps]].  You can override any of the default
+ * properties with passing in your own properties.
+ *
+ * @param topic Kafka topic to produce to.
+ * @param brokers Kafka Broker metadata list.
+ * @param properties Producer properties.
+ * @tparam K Kafka "Key" type
+ * @tparam M Kafka "Message" type
+ * @tparam KE Encoder for K. Must have a constructor argument signature of `properties: VerifiableProperties = null`
+ * @tparam ME Encoder for M. Must have a constructor argument signature of `properties: VerifiableProperties = null`
+ */
+class GenericKafkaProducer[K,M,KE <: Encoder[K]: Manifest,ME <: Encoder[M]: Manifest](topic: String,
+                                                                                      brokers: List[AddressAndPort] = List.empty,
```

Comment:
Yeah silly Kafka.  This is kind of why I thought having a consul client embedded in the library would make sense if users chose to use it or not.  It is not terribly crazy because the client would essentially be curling against the consul agent at localhost.